File: Application/Curricula/GetCurricula/GetCurriculaQuery.cs
using System.Collections.Generic;
using System.Linq;
using System.Threading;
using System.Threading.Tasks;
using Biobrain.Application.Common.Core;
using Biobrain.Application.Interfaces.DataAccess;
using JetBrains.Annotations;
using Microsoft.EntityFrameworkCore;

namespace Biobrain.Application.Curricula.GetCurricula
{
    [PublicAPI]
    public class GetCurriculaQuery : ICommand<List<GetCurriculaQuery.Result>>
    {
        [PublicAPI]
        public class Result
        {
            public int CurriculumCode { get; set; }
            public string Name { get; set; }
        }


        internal class Handler : CommandHandlerBase<GetCurriculaQuery, List<Result>>
        {
            public Handler(IDb db) : base(db)
            {
            }

            public override async Task<List<Result>> Handle(GetCurriculaQuery request, CancellationToken cancellationToken)
            {
                return await Db.Curricula
                               .Select(_ => new Result
                                            {
                                                CurriculumCode = _.CurriculumCode,
                                                Name = _.Name
                                            })
                               .ToListAsync(cancellationToken);
            }
        }
    }
}

File: BiobrainWebAngular/src/app/api/curricula/get-curricula.query.ts
import { ApiPath } from '../api-path.service';
import { Query } from '../common/query';

export class GetCurriculaQuery extends Query<GetCurriculaQuery_Result[]> {
  getUrl(apiPath: ApiPath): string {
    return `${apiPath.curricula()}/GetCurricula`;
  }
}

export interface GetCurriculaQuery_Result {
  curriculumCode: number;
  name: string;
}
